### PR TITLE
Add support for Log4j2

### PR DESCRIPTION
**Pull request by mridang**

### CODE REVIEW

File: framework/src/play/Logger.java [hunk 1]
```diff
@@ -105,7 +117,10 @@ public static void setUp(String level) {
         if (forceJuli || log4j == null) {
             Logger.juli.setLevel(toJuliLevel(level));
         } else {
-            Logger.log4j.setLevel(org.apache.log4j.Level.toLevel(level));
+            LoggerContext ctx = (LoggerContext) LogManager.getContext(false);
+            LoggerConfig loggerConfig = ctx.getConfiguration().getLoggerConfig(Logger.juli.getName());
+            loggerConfig.setLevel(org.apache.logging.log4j.Level.OFF);
```
Comment (oostman) [inline framework/src/play/Logger.java]: Should it not set level instead of always using Level#OFF?

**Comment by oostman** [comment]
Looks good, left one comment

File: framework/src/play/Logger.java [hunk 2]
```diff
@@ -65,31 +72,36 @@ public static void init() {
             log4jConf = Logger.class.getResource(log4jPath);
         }
         if (log4jConf == null) {
-            Properties shutUp = new Properties();
-            shutUp.setProperty("log4j.rootLogger", "OFF");
-            PropertyConfigurator.configure(shutUp);
+            LoggerContext ctx = (LoggerContext) LogManager.getContext(false);
+            LoggerConfig loggerConfig = ctx.getConfiguration().getRootLogger();
+            loggerConfig.setLevel(org.apache.logging.log4j.Level.OFF);
```
Comment (leandroutn) [inline framework/src/play/Logger.java]: what does this line do?
Sets the default and lets the configuration define the each particular level?

File: framework/src/play/utils/ANSIConsoleAppender.java [hunk 3]
```diff
@@ -1,67 +0,0 @@
-package play.utils;
-
-import org.apache.log4j.ConsoleAppender;
-import org.apache.log4j.Priority;
-import org.apache.log4j.spi.LoggingEvent;
-
-/**
- * Colour-coded console appender for Log4J.
- */
-public class ANSIConsoleAppender extends ConsoleAppender {
```
Comment (leandroutn) [inline framework/src/play/utils/ANSIConsoleAppender.java]: this was not used at all?

**Comment by leandroutn** [comment]
looks good to me too (haven't tested it though)

File: framework/src/play/utils/ANSIConsoleAppender.java (hunk 3 above)
Comment (mridang) [inline framework/src/play/utils/ANSIConsoleAppender.java]: It was but it is now built into Log4J2. This is redundant.

File: framework/src/play/Logger.java (hunk 2 above)
Comment (mridang) [inline framework/src/play/Logger.java]: If not logging configuration is defined. It turns it off entirely.

**Pull request merged**